Markup/samples/js/custom.js

```diff
@@ -22,8 +22,207 @@ $(document).ready(function () {
+        showPopoutIcon: false,
+        showMaximiseIcon: false,
+        showCloseIcon: false,
+    },
+    dimensions: {
+        borderWidth: 5,
+        minItemHeight: 10,
+        minItemWidth: 0,
+        headerHeight: 20,
+        dragProxyWidth: 300,
+        dragProxyHeight: 200,
+    },
+    labels: {
+        close: 'close',
+        maximise: 'maximise',
+        minimise: 'minimise',
+        popout: 'open in new window',
+    },
+
+    content: [{
+        type: 'row',
+        width: 100,
+        content: [
+            {
+                id: "actionContainer",
+                type: 'component',
+                componentName: 'actionContainer',
+                componentState: {
+                    text: 'Component 1',
+                    color: '#fff',
+                },
+                title: "Action",
+                width: 100
+            },
+            {
+                id: "viewContainer",
+                title: "title",
+                type: 'component',
+                componentName: 'viewContainer',
+                componentState: {text: "text"},
+                width: 0
+            }
+        ]
+    }]
+};
+
+var myLayout = new GoldenLayout(config, '#wrapper');
+
+myLayout.registerComponent('actionContainer', function (container, componentState) {
+    container.getElement().html($("#actionContainer"));
+});
+
+myLayout.registerComponent('viewContainer', function (container, componentState) {
+    container.getElement().html($("#viewContainer"));
+});
+
+myLayout.init();
+
+var toggleRowColumn = function () {
+    var oldElement = myLayout.root.contentItems[0],
+        newElement = myLayout.createContentItem({
+            type: oldElement.isRow ? 'column' : 'row',
+            content: []
+        });
+
+    //Prevent it from re-initialising any child items
+    newElement.isInitialised = true;
+
+    for (i = 0; i < oldElement.contentItems.length; i++) {
+        newElement.addChild(oldElement.contentItems[i]);
+    }
+    myLayout.root.replaceChild(oldElement, newElement);
+
+    var actionContainer = myLayout.root.contentItems[0].contentItems[0];
+    var viewContainer = myLayout.root.contentItems[0].contentItems[1];
+
+    if (newElement.isColumn) {
+        $('#toggleLayout').html('<span data-toggle="tooltip" data-placement="bottom" title="Dock to right">' +
+            '<i class="fas fa-columns" ></i></span>');
+        columnLayout(actionContainer, viewContainer);
+    } else {
+        $('#toggleLayout').html('<span data-toggle="tooltip" data-placement="bottom" title="Dock to bottom">' +
+            '<i class="fas fa-window-maximize"></i></span>');
+        rowLayout(actionContainer, viewContainer);
+    }
+
+    myLayout.updateSize();
+};
+
+$('.request-response-title').on("click", function (e) {
+    var display = $(this).next().css('display');
+    $(this).next().slideToggle();
+    if (display == "block") {
+        $(this).children().children("i").removeClass("fa-angle-right");
```

Comment:
Please use a class name and select the child instead of repeating the `.children()` jquery API